evaluators/text/BleuEvaluator/__init__.py

```diff
@@ -7,17 +7,48 @@ class BleuEvaluator(BaseTextEvaluator):
     It will use the Bleu on NLTK package.
     A perfect match will score 1.0 and a perfect unmatch will score 0.0
     """
+
+    def count_ngram(self, text, max_ngram):
+        """Gets all n_grams from a given text
+            - text: the text to be analyzed
+            - max_ngram: max n_gram from the text
+            - counter: dictionary of n-gram and its count
+        """
+        from collections import Counter
+
+        counter = Counter()
+
+        for n_order in range(1, max_ngram):
+            for i in range(len(text) - n_order):
+                ngram = tuple(text[i:i+n_order])
+                counter[ngram] += 1
+        
+        return counter
+
     
-    def get_nltk_bleu_score(self, desired, actual):
+    def get_nltk_bleu_score(self, desired, actual, overlap):
         import nltk.translate.bleu_score as bleu    
         from nltk.translate.bleu_score import SmoothingFunction
 
-        return bleu.sentence_bleu(desired, actual, smoothing_function=SmoothingFunction().method4)
+        #Check the n-gram and reset the weights accordingly
+        overlaps = sum(overlap.values())
+        if overlaps == 1:
+            return bleu.sentence_bleu(desired, actual, weights = (1, 0, 0, 0), smoothing_function=SmoothingFunction().method4)
+        if overlaps == 2:
+            return bleu.sentence_bleu(desired, actual, weights = (0, 1, 0, 0), smoothing_function=SmoothingFunction().method4)
+        if overlaps == 3:
+            return bleu.sentence_bleu(desired, actual, weights = (0, 0, 1, 0), smoothing_function=SmoothingFunction().method4)
+        if overlaps == 4:
+            return bleu.sentence_bleu(desired, actual, weights = (0, 0, 0, 1), smoothing_function=SmoothingFunction().method4)
+        else:
+            return bleu.sentence_bleu(desired, actual) #otherwise use standard weights (0.25, 0.25, 0.25, 0.25) and no Smoothin function
 
 
     def evaluate(self,
             actual,
             desired,
+            max_order=4,
```

Comment:
These parameters do not belong here, they may be part of  the `__init__`. but please do not change the `interface`